src/services/hsm.service.js

```diff
@@ -141,78 +361,57 @@ async function derivateHtrCkd(
     }
   }
 
-  // Derivation 1: Bip Code
-  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
-    derivationVersion, // Version
-    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 44, // Derivation index
-    false, // Exportable
-    true, // Temporary
-    hsmKeyName, // Parent key name
-    childKeyNames.HTR_CKD_BIP_KEYNAME, // Child key name
-  );
+  if (!acctPathExists) {
+    // derive from root to account path
+    const childKeyNames = {
+      HTR_CKD_BIP_KEYNAME: 'HTR_BIP_KEY',
+      HTR_CKD_COIN_KEYNAME: 'HTR_COIN_KEY',
+    };
 
-  // Derivation 2: Coin
-  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
-    derivationVersion,
-    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 280,
-    false,
-    true,
-    childKeyNames.HTR_CKD_BIP_KEYNAME,
-    childKeyNames.HTR_CKD_COIN_KEYNAME,
-  );
+    // Derivation 1: Bip Code
+    await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+      derivationVersion, // Version
+      hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 44, // Derivation index
+      false, // Exportable
+      true, // Temporary
+      hsmKeyName, // Parent key name
+      childKeyNames.HTR_CKD_BIP_KEYNAME, // Child key name
+    );
 
-  // Derivation 3: Account
-  await hsmConnection.blockchain.createBip32ChildKeyDerivation(
-    derivationVersion,
-    hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE,
-    false,
-    true,
-    childKeyNames.HTR_CKD_COIN_KEYNAME,
-    childKeyNames.HTR_CKD_ACCOUNT_KEYNAME,
-  );
+    // Derivation 2: Coin
+    await hsmConnection.blockchain.createBip32ChildKeyDerivation(
+      derivationVersion,
+      hsm.enums.BCHAIN_SECURE_BIP32_INDEX.BASE + 280,
```

Comment:
We have 280 as a constant on the lib, don't we?

Reply:
Changed to use this constant